internal/stores/metadata.go

```diff
@@ -398,7 +429,12 @@ func (s *SQLStore) SetContractSet(ctx context.Context, name string, contractIds
 }
 
 func (s *SQLStore) RemoveContract(ctx context.Context, id types.FileContractID) error {
-	return removeContract(s.db, fileContractID(id))
+	err := removeContract(s.db, fileContractID(id))
+	if err != nil {
+		return err
+	}
+	delete(s.isOurContract, id)
```

Comment:
do we have cascaded deletes to worry about here? I guess we don't

Reply:
I really hope we don't delete contracts implicitly as the result of a cascade. That would be quite bad.